server/app/controllers/applicant/ApplicantProgramsController.java

```diff
@@ -215,4 +225,16 @@ public CompletionStage<Result> edit(Request request, long applicantId, long prog
   private static Result redirectToHome() {
     return redirect(controllers.routes.HomeController.index().url());
   }
+
+  private Optional<Long> getApplicantId(Request request) {
+    CiviFormProfileData profileData =
+        profileUtils.currentUserProfile(request).orElseThrow().getProfileData();
+
+    if (profileData == null) {
+      return Optional.empty();
+    }
+
+    return Optional.ofNullable(
+        profileData.getAttribute(ProfileFactory.APPLICANT_ID_ATTRIBUTE_NAME, Long.class));
+  }
```

Comment:
suggestion (blocking): this will be reused in future PRs that handle the other applicant routes, consider moving somewhere shared such as the parent controller class or `ProfileUtils`

Reply:
Done